### DISCUSSION TITLE
Update field value on submit before saving row, value can't be blank

### DISCUSSION THREAD

**Discussion by chiptech-mitchell** [Budibase/budibase#8833] [Help]
I have created a form that saves to the budibase db, there is a required field (number) and this field needs to be populated based on the execution of a query.

My submit button actions are 

1. execute query
2. update value
3. save row

![image](https://user-images.githubusercontent.com/109489679/204169341-2048a0d3-6059-4120-ab47-12de1292af9f.png)

however when I do this I get the error message field can't be blank, however if I give the number field component a default value the row is saved with the default value and the number field is updated to the value given by the `update value` action and then if I submit the form again the number field which has the result of the execute query "on submit" is saved as I want.

My question is how can I either get the query result to save on the first go, or can I prepopulate the number field with the query result before I hit the submit.

Thanks in advance

**Comment by andz-bb** [comment]
hey @chiptech-mitchell, I created a similar set up on my end and am seeing the same issue. The actions are currently being run asynchronously so the Save Row action runs before the field is updated to the value from the query. There's an open issue requesting an option to run the actions synchronously here #6872 - it's worth adding your interest there so that the development can be prioritised.

**Reply by chiptech-mitchell** [reply]
Oh I see, ok yea im trying to figure out a way around that, but I havent had much luck yet
